rero_ils/modules/persons/api.py

```diff
@@ -161,3 +161,20 @@ def _get_author_for_document(self):
         if variant_person:
             author['variant_name'] = unique_list(variant_person)
         return author
+
+    @property
+    def organisation_pids(self):
+        """Get organisations pids."""
+        organisations = []
+        search = DocumentsSearch().filter('term', authors__pid=self.pid)
+        agg = A('terms',
+                field='holdings.organisation.organisation_pid', size=10)
+        search.aggs.bucket('organisation', agg)
+        results = search.execute()
+        for result in results:
+            hold_pids = Holding.get_holdings_pid_by_document_pid(result['pid'])
+            for hold_pid in hold_pids:
+                organisations.append(
+                    Holding.get_record_by_pid(hold_pid).organisation_pid
+                )
+        return list(set(organisations))
```

Comment:
why you doesn't use the result of aggregation to get organisation pid ?
Something like :
```python
results = search.execute()
for result in results.aggregations.organisation.buckets :
  if result.doc_count > 0 :
    organisations.append(result.key)
return organisations
```

Reply:
Because I did not know how to do it.